android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java

```diff
@@ -92,26 +91,20 @@ protected int getLayoutResource() {
     private void setupViewPager(final ViewPager viewPager) {
         adapter = new ScheduleViewPagerAdapter(getChildFragmentManager());
 
-        realmRepo.getEventDates()
-                .addChangeListener((eventDates, orderedCollectionChangeSet) -> {
-                    int eventDays = eventDates.size();
-                    for (int i = 0; i < eventDays; i++) {
-                        String date = eventDates.get(i).getDate();
-
-                        try {
-                            adapter.addFragment(new DayScheduleFragment(),
-                                    DateConverter.formatDay(date), date);
-                            adapter.notifyDataSetChanged();
-                        } catch (DateTimeParseException pe) {
-                            Timber.e(pe);
-                            Timber.e("Invalid date %s in database", date);
-                        }
-                    }
-                });
+        scheduleFragmentViewModel.getEventDatesString().observe(this, eventDatesString -> {
+            int eventDays = eventDatesString.size();
+            for (int i = 0; i < eventDays; i++) {
+                String date = eventDatesString.get(i).first;
+                String formattedDate = eventDatesString.get(i).second;
+                adapter.addFragment(new DayScheduleFragment(),
+                        formattedDate, date);
+                adapter.notifyDataSetChanged();
+            }
+        });
 
-        realmRepo.getTracks().addChangeListener((tracks, orderedCollectionChangeSet) -> {
+        scheduleFragmentViewModel.getTracks().observe(this,tracksList -> {
             this.tracks.clear();
-            this.tracks.addAll(tracks);
+            this.tracks.addAll(tracksList);
             tracksNames = new String[this.tracks.size()];
             isTrackSelected = new boolean[this.tracks.size()];
```

Comment:
These can be moved to ViewModel as well

Reply:
They are arrays. I don't think they have an ```addAll``` method. Hence i think we will still need a ```for``` loop.

Reply:
I meant these variables should be moved to ViewModel

```
tracksNames = new String[this.tracks.size()];
isTrackSelected = new boolean[this.tracks.size()];
```

But let it be, the implementation should be changed and new issue should be created for changing the filter functionality

Reply:
Alright, I will create an issue for the filter functionality